addons/mail/static/tests/suggestion/suggestion_tests.js

```diff
@@ -249,3 +249,34 @@ QUnit.test("Internal user should be displayed first", async (assert) => {
     assert.strictEqual(suggestions[2].textContent, "Person B(b@test.com)");
     assert.strictEqual(suggestions[3].textContent, "Person A(a@test.com)");
 });
+
+QUnit.test("Current user that is a follower should be considered as such", async (assert) => {
+    const pyEnv = await startServer();
+    const userId = pyEnv["res.users"].create({ share: true });
+    const partnerIds = pyEnv["res.partner"].create([
+        { email: "a@test.com", name: "Person A" },
+        { email: "b@test.com", name: "Person B", user_ids: [userId] },
+    ]);
+    pyEnv["mail.followers"].create([
+        {
+            is_active: true,
+            partner_id: partnerIds[1], // B
+            res_id: pyEnv.currentPartnerId,
+            res_model: "res.partner",
+        },
+        {
+            is_active: true,
+            partner_id: pyEnv.currentPartnerId,
+            res_id: pyEnv.currentPartnerId,
+            res_model: "res.partner",
+        },
+    ]);
+    const { openFormView } = await start();
+    await openFormView("res.partner", pyEnv.currentPartnerId);
+    await click("button:contains(Send message)");
+    await insertText(".o-mail-Composer-input", "@");
+    const suggestions = document.querySelectorAll(".o-mail-Composer-suggestion");
+    assert.strictEqual(suggestions[0].textContent, "Mitchell Admin");
+    assert.strictEqual(suggestions[1].textContent, "Person B(b@test.com)");
+    assert.strictEqual(suggestions[2].textContent, "Person A(a@test.com)");
```

Comment:
need to use `contains` (with before/after if you need order), querySelector is prone to race conditions